sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -585,34 +589,6 @@ private[hive] class HiveMetastoreCatalog(hive: HiveContext) extends Catalog with
         }
 
       case p: LogicalPlan if p.resolved => p
-
-      case p @ CreateTableAsSelect(db, tableName, child, allowExisting, None) =>
```

Comment:
For data source table, it does not use `CreateTableAsSlect` because it has its own semantics like `provider`, `temporary`, and `mode`, and `options`. So, i think it is better to have its own logical plan instead of mixing these in `CreateTableAsSlect`.

For `hasStorageSpec`, the name can be better. The explanation of hive.convertCTAS can be found at https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveContext.scala#L72.
It is mainly used to avoid saving data to text and sequence file. For now, Hive will not set serde when text file or sequence file is used, i.e. `crtTbl.getSerName` will be null (see `fillStorageFormat` and `fillDefaultStorageFormat`).